gap/oper.gi

```diff
@@ -1891,3 +1891,31 @@ function(filt, digraph)
   fi;
   return AsSemigroup(IsPartialPermSemigroup, digraph);
 end);
+
+InstallMethod(IsDigraphColouring, "for a digraph and a list",
+[IsDigraph, IsHomogeneousList],
+function(digraph, colours)
+  local n, out, v, w;
+  n := DigraphNrVertices(digraph);
+  if Length(colours) <> n or (n > 0 and not IsPosInt(colours[1])) then
```

Comment:
I know you're not finished with, but just in case it's useful, here's a comment:
```
gap> IsHomogeneousList([1,-1]);
true
```
So just because the first point in `colours` is a positive integer, it doesn't mean that the rest of the integers in `colours` are positive.

Reply:
Good point :)